pandas/core/generic.py

```diff
@@ -4375,7 +4375,8 @@ def __getattr__(self, name):
                 name in self._accessors):
             return object.__getattribute__(self, name)
         else:
-            if name in self._info_axis:
```

Comment:
> e.g. a string in a DTI

These can't be valid identifiers though, so they by definition can't be going through `__getatrr__`.

We can perform this cheap check at the *class* level, without having to deal with values at all.

Reply:
Note that this isn't addressing the other performance problems with DTI, which are left for https://github.com/pandas-dev/pandas/issues/17754

Reply:
as I said, this is not a good solution here. Call a method on the object. In the future folks will have no idea this is called, nor where it is, it add unecessary context and complexity.

Reply:
> In the future folks will have no idea this is called

We have a comment explaining it on the base Index class and a link back to the original issue.

In what case will this cause issues? It's short-circuting a check that will always return False.

Reply:
his is just technical debt. pls just use a function, its much much cleaner.

Reply:
Wait, your objection is to the use of a class attribute instead of a function? Why'd you say "not a good solution here" and bring up things about the hash table?

Reply:
no many objection is to the if logic inside the getattribute itself 

it is so buried

Reply:
Agreed with Joris. This if block is specifically for dot selection. It makes sense to be there.

Reply:
my point is that you instead should be calling a function like

```
if self._info_axis._._can_hold(name):
     return self[name]
```

putting this here buries all kinds of logic in an obscure routine. no-one will be able to understand / find this later.
These should be the simplest routines possible, while pushing logic to the axis object itself.

Reply:
Why would it be a function? To outcome doesn't depend on `name`.

> putting this here buries all kinds of logic in an obscure routine

Can you say *exactly* what logic you think I'm burying here? You have `if self._info_axis._can_hold(name)`. I have `if self._info_axis._can_hold_identifiers`.

Reply:
you are checking for a random attribute on an index class. This is so fragile, with no documentation whatsoever. Simply make it a function call, I am not sure why this is that hard.

Reply:
> I am not sure why this is that hard.

Uhm...